check file before reading

aowen87 opened the pull request:
### Description
I'm adding in a simple file check to the mili reader. Rapidjson tends to hang when it tries to open a file that is not in json format. This check will prevent this hang. This is mostly a concern with people transitioning from the old mili format to the new one. 

I also changed the name of makemili to makemili_driver so that the name is consistent with the command line help that is displayed, and I've removed the old makemili code. makemili_driver now lives in visit/src/bin/. 


### How Has This Been Tested?
I've re-run the mili tests, and everything passes. I've also tested trying to open a mili file that is not in the correct format, and it doesn't hang anymore. 

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the release notes
- [ ] I have made corresponding changes to the documentation
- [x] I have added debugging support to my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added any new baselines to the repo
- [ ] I have assigned reviewers (see [VisIt's PR procedures](https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/pr_create.html#reviewers) for more information).

aowen87 committed:
check file before reading

aowen87 committed:
removing old makemili

biagas commented on src/databases/Mili/avtMiliFileFormat.C:
```diff
@@ -3464,15 +3464,34 @@ avtMiliFileFormat::ExtractJsonClasses(rapidjson::Document &jDoc,
+        sprintf(msg, "Invalid Mili file. You are likely using an outdated "
+            "format. To update your format, use the makemili_driver located "
+            "in the visit/src/bin directory. ");
+        EXCEPTION1(InvalidFilesException, fpath);
```
Are you sure you want to reference 'src' in the output.  It won't be relevant from an installed VisIt.

aowen87 replied:
ah, good point... I forgot about that bit. Maybe I should have visit copy makemili_driver to the exe directory.

aowen87 replied:
Okay, I updated the the build to copy the makemili_driver to exe, and I updated that comment. How does that look?

aowen87 committed:
adding makemili_driver to exe

aowen87 committed:
updating comment

aowen87 committed:
updating comment and adding debug

markcmiller86 commented:
@aowen87 have a look at examples of use of NonCompliantFileException to get your error message about running makemili_driver back to the screen. I think if you do business that way, it should work. Another option...look at use of avtWarningCallback::IssueWarning as well. Hope that helps.

aowen87 commented:
Awesome, I'll check that out. Thanks, Mark.

biagas commented:
I think I am a bit confused. makemili_driver is a binary file residing in visit/src/bin, correct?
It isn't something that will be generated with VisIt when VisIt is compiled? 
Is there anything in /src/tools/data/makemili besides the CMakeLists.txt? If not, that directory isn't needed.

If you want makemili_driver copied to the build dir and installed when we create the release package, there is logic in /src/bin/CMakeLists.txt file that can be followed.  It shouldn't be installed for Windows.

aowen87 commented:
Yeah, I had a misunderstanding about how the visit installs work. I had been mimicking the pattern of the old makemili, thinking that exe files are copied over to installs: I put makemili_driver in src/tools/data and bin and had the CMakeLists file in src/tools/data just copy over makemili_driver to exe. 

It probably makes more sense to just have makemili_diver live in bin and add it to the list of install files for non-windows builds. I'll go ahead and do that when I'm back on VisIt next week.

aowen87 committed:
updating makemili install and file error message

aowen87 commented:
I just updated with the changes I mentioned last week. 
1. makemili no longer lives in tools/data. Instead, it only resides in bin, and it is copied over when installed. 
2. I added a change to the type of file exception that is raised. The error message now reports appropriately.

aowen87 committed:
Merge branch '3.0RC' into task/maguire7/2019_07_mili_plugin_safety_read
